drivers/media/platform/raspberrypi/pisp_be/pisp_be.c

```diff
@@ -303,51 +270,50 @@ static void hw_queue_job(struct pispbe_dev *pispbe,
 	 * and we don't want to modify (or be vulnerable to modifications of)
 	 * the mmap'd buffer.
 	 */
-	for (u = 0; u < N_HW_ADDRESSES; ++u) {
-		write_reg(pispbe, PISP_BE_IO_INPUT_ADDR0_LO_OFFSET + 8 * u,
-			  (u32)(hw_dma_addrs[u]));
-		write_reg(pispbe, PISP_BE_IO_INPUT_ADDR0_LO_OFFSET + 8 * u + 4,
-			  (u32)(hw_dma_addrs[u] >> 32));
+	for (unsigned int u = 0; u < N_HW_ADDRESSES; ++u) {
+		pispbe_wr(pispbe, PISP_BE_IO_INPUT_ADDR0(u),
+			  lower_32_bits(job->hw_dma_addrs[u]));
+		pispbe_wr(pispbe, PISP_BE_IO_INPUT_ADDR0(u) + 4,
```

Comment:
See earlier comment at L63 - the macro shouldn't be called `INPUT`...`0` if it doesn't refer to input component 0.
And the + 4 isn't nice.

Reply:
I agree with all but `INPUT` as that's the name used in specs

Reply:
oh, now I see, it's not only used for `INPUT` but for all addresses. I'll remove it!